src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -1008,6 +1022,24 @@ private void FlushOpenTelemetry(IHost host)
             }
         }
 
+        private ScriptHostStartupOperation BeginStartupOperation(CancellationToken parentToken, Guid? parentId = null)
+        {
+            var operation = new ScriptHostStartupOperation(parentToken, parentId);
+            _activeStartupOperations.TryAdd(operation, byte.MinValue);
```

Comment:
Do we want any special handling here when this fails? (main thing in mind is the `parentId` set `null` case)

Reply:
I can't think of why it'd fail as the key here is the object itself. If this object already existed in our tracking collection, then we'd still be in an okay state.

This behavior didn't really change with this; it originally lived inside the `ScriptHostStartupOperation` type itself (and static). So it's been like this for quite a while.